### PR TITLE
Fix io.Closer support for connectors

### PR DESCRIPTION
**Pull request by begelundmuller**
It appears that closing a DB using `XSAM/otelsql` does not always propagate to the underlying driver. That's apparently due to `io.Closer` being optional for connectors and `database/sql` using a type assertion to check whether the connector implements it. That type assertion does not pass through to `otConnector.Connector`, so we need to explicitly implement `Close() error` on `otConnector` and replicate the type assertion there.

### CODE REVIEW

File: connector.go [hunk 1]
```diff
@@ -62,6 +63,15 @@ func (c *otConnector) Driver() driver.Driver {
 	return c.otDriver
 }
 
+func (c *otConnector) Close() error {
+	// database/sql uses a type assertion to check if connectors implement io.Closer.
+	// The type assertion does not pass through to otConnector.Connector, so we explicitly implement it here.
+	if closer, ok := c.Connector.(io.Closer); ok {
```
Comment (XSAM) [inline connector.go:69]: Could you please add an implement assertion in line 26 to show `otConnector` implements `io.Closer`?
https://github.com/XSAM/otelsql/pull/199/files#diff-c1caa8de39e277339469495d65efd1a363c54065c3cc02e7422b097774e56c86R25
It should be

```go
var _ io.Closer = (*otConnector)(nil)
```

**Review by XSAM — COMMENTED** [review]
Hi @begelundmuller, thanks for raising this PR. It looks good.

Please also update the `CHANGELOG.md` to reflect this fix.

**Comment by begelundmuller** [comment]
@XSAM thanks for the rapid review! I have addressed both of your comments.

**Review by XSAM — APPROVED** [review]

**Comment by XSAM** [comment]
@begelundmuller Thanks for your contribution.

Please update this branch with the base branch.

**Comment by begelundmuller** [comment]
> @begelundmuller Thanks for your contribution.
> 
> Please update this branch with the base branch.

Done!

**Pull request merged**